docs/DeveloperGuide.md

```diff
@@ -195,38 +195,53 @@ The following sequence diagram shows how a list is autosorted upon an addition o
     * Pros: Relatively low time complexity i.e. O(logn).
     * Cons: Prone to error and difficult to implement.
 
-### [In progress] Switch Mode feature
+### Switch Mode feature
+
+Productiv can be one of these modes: dashboard, deliverable, meeting and contact mode.
+Based on the current mode, user input is passed to the corresponding `LogicManager`,
+e.g. if the user is in deliverable mode, user input is passed to `LogicDeliverableManager`.
 
 #### Implementation
 
-Productiv can be in different modes: dashboard, deliverable, meeting and contact mode. 
-Based on the current mode, the user input is passed to the relevant `LogicManager`. 
-Following that, the `LogicManager` will parse the user input and produce the relevant results.
-The current mode is represented by a `ModeEnum` and stored in `MainWindow`.
+The user input is handled and retrieved by the `MainWindow` and then passed to `LogicModeManager`.
+`LogicModeManager` will call `ModeParser`, which will parse the input and create a `SwitchCommandParser`. `SwitchCommandParser` will return a `SwitchCommand`.
+`LogicModeManager` will then execute `SwitchCommand` which returns a `CommandResult` containing the mode that the app should switch to. 
+Then, `MainWindow` gets the new mode to switch to from `CommandResult`.
+Based on the mode, `MainWindow` will update its own attribute `mode`.
+`MainWindow` will then update the UI to show only related information to the new mode.
```

Comment:
Can paraphrase to "...update the UI to only show information related to the new mode."